File: app/src/main/java/com/example/mvctutorial/designpattern/_07builder/Mac.java
package com.example.mvctutorial.designpattern._07builder;

public class Mac extends BluePrint{

    private Computer computer;

    public Mac() {
        this.computer = new Computer("","","");
    }

    @Override
    void setCpu() {
        computer.cpu = "m1 pro max";
    }

    @Override
    void setRam() {
        computer.ram = "32g";
    }

    @Override
    void setStorage() {
        computer.storage = "1T ssd";
    }

    @Override
    Computer getComputer() {
        return computer;
    }

}

File: app/src/main/java/com/example/mvctutorial/designpattern/_07builder/Mac.kt
package com.example.mvctutorial.designpattern._07builder

class Mac: BluePrint() {

    private var computer = Computer("", "", "")

    override fun setCpu() {
        computer.cpu = "m1 pro max"
    }

    override fun setRam() {
        computer.ram = "32g"
    }

    override fun setStorage() {
        computer.storage = "1T ssd"
    }

    override fun getComputer(): Computer {
        return computer
    }
}